In [1]:
import astropy.units as u
import astropy.constants as const
import numpy as np

# Use to obtain data
import pandas as pd
import urllib
# the service URL
livechart = "https://nds.iaea.org/relnsd/v1/data?"

# There have been cases in which the service returns an HTTP Error 403: Forbidden
# use this workaround
import urllib.request
def lc_pd_dataframe(url):
    req = urllib.request.Request(url)
    req.add_header('User-Agent', 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:77.0) Gecko/20100101 Firefox/77.0')
    return pd.read_csv(urllib.request.urlopen(req))

## Quesiton 1

Assume $\rm^{235}U$ splits into two stable nuclides $\rm^{100}Ru$, $\rm^{132}Xe$, and three neutrons. About 86\% of the energy is in the kinetic energy of the fission fragments. A fully enriched $\rm^{235}U$ bomb is expected to give a prompt explosive yield of about $20\,\rm{kg}$ of TNT. How many kilograms of $\rm^{235}U$ are required (assuming the actual explosive yield efficiency is around 30\%)?

### Answer

The energy of this reaction have been calculated in the previous question. I copied the result here.

$$
\Delta E = 192\,{\rm MeV}
$$

In [3]:
m_235U = lc_pd_dataframe(livechart+"fields=ground_states&nuclides=235U")['atomic_mass'][0]*1e-6*u.u
m_100Ru = lc_pd_dataframe(livechart+"fields=ground_states&nuclides=100Ru")['atomic_mass'][0]*1e-6*u.u
m_132Xe = lc_pd_dataframe(livechart+"fields=ground_states&nuclides=132Xe")['atomic_mass'][0]*1e-6*u.u
m_n = lc_pd_dataframe(livechart+"fields=ground_states&nuclides=1n")['atomic_mass'][0]*1e-6*u.u
m_e = (const.m_e).to('u')
Delta_m = m_235U - (6*m_e + 3*m_n + m_100Ru + m_132Xe)
Delta_E = (Delta_m*const.c**2).to("MeV")
Delta_Ep = Delta_E*0.86
Delta_Ep

<Quantity 165.24487211 MeV>

And the energy convert to explosion $\Delta E' = 0.86\Delta E = 165\,{\rm MeV}$.

Accoriding to Wikipedia, a ton of TNT can release about 4.184 gigajoules. So, the energy of explosion of $20\,{\rm kg}$ is $E_0$ where

$$
E_0 = 4.184\,{\rm GJ} \cdot \frac{20\,{\rm kg}}{1\,{\rm t}}
$$
$$

In [14]:
E0 = (4.184*u.GJ*20*u.kg/u.t).to('MJ')
E0

<Quantity 83.68 MJ>

Thus $E_0 = 83.68\,{\rm MJ}$.

The number of $\rm ^{235}U$ required is $N = E_0/{\Delta E'}$.

In [18]:
N = (E0/Delta_Ep).si
N

<Quantity 3.16070007e+18>

So $N =3.16\times 10^{18}$. The mass of $\rm ^{235}U$ is

$$
m = N\cdot m_{\rm ^{235}U}
$$

In [22]:
m = (N*m_235U).to('mg')
m

<Quantity 1.23362005 mg>

So the mass of $\rm ^{235}U$ is $1.23\,{\rm mg}$.

## Question 2

Describe two methods used for separating 235U from natural uranium.

### Answer

Two methodologies are employed for isolating $\rm ^{235}U$ from its natural form, each initiating with gaseous uranium hexafluoride as the primary substrate.

1. **Gaseous Diffusion:** This process subjects the gas to a sequence of semipermeable membranes. Through thousands of membrane passages, the concentration of $^{235}$U incrementally escalates.

2. **Centrifugation Technique:** Involves introducing the gas into ultracentrifuges, which rotate at an incredibly high pace—thousands of revolutions per minute. Similar to gaseous diffusion, repeated cycles lead to a stepwise augmentation in $\rm ^{235}U$ concentration. 

Both techniques, albeit distinct in their operational mechanics, share the objective of progressively enriching $\rm ^{235}U$ via iterative processes.

## Question 3

What is the meaning of critical mass or critical size? How to roughly estimate the
critical size of $\rm ^{233}U$ bomb?

### Answer

Critical mass is the smallest amount of fissile material needed for a sustained nuclear
chain reaction.
Critical size is the least include enough fissionable material to reach critical mass.

The fission cross section of $\rm ^{233}U$ is $\sigma_f = 1.9\,{\rm b}$. We know that the density of uranium is $19\,{\rm g/cm^3}$. Thus the mean free path is

$$
\lambda  = \frac 1{\sigma_fN} = \frac{m_{\rm ^{233}U}}{\sigma_f \rho}
$$


In [36]:
sigma_f= 1.9*u.barn
rho = 19*u.g/(u.cm**3)

lam = (m_235U/(sigma_f*rho)).si
lam

<Quantity 0.10811624 m>

Thus the critical size is $R_c = \lambda/2$ and the critical mass is $m_c = \frac{4\pi R_c^3}3 \rho$.

In [38]:
R_c = lam/2
display(R_c)
m_c = (4*np.pi*R_c**3/3*rho).to(u.kg)
m_c

<Quantity 0.05405812 m>

<Quantity 12.57259771 kg>

So the critical size is $0.05\,{\rm m}$ and the critical mass is $12.6\,{\rm kg}$.